hera_cal/utils.py

```diff
@@ -845,6 +845,44 @@ def chisq(data, model, data_wgts=None, gains=None, gain_flags=None, split_by_ant
     return chisq, nObs, chisq_per_ant, nObs_per_ant
 
 
+def gain_relative_difference(old_gains, new_gains, flags, denom=None):
+    """Compuate relative gain differences between two sets of calibration solutions 
+    (e.g. abscal and smooth_cal), as well as antenna-averaged relative gain differences.
+    
+    Arguments:
+        old_gains: dictionary mapping keys like (0, 'Jxx') to waterfalls of complex gains.
+            Must contain all keys in new_gains.
+        new_gains: dictionary mapping keys like (0, 'Jxx') to waterfalls of complex gains
+        flags: dictionary mapping keys like (0, 'Jxx') to boolean flag waterfalls. Must
+            contain all keys in new_gains.
+        denom: gain dictionary to use to normalize the relative difference. Default None
+            uses old_gains.
+
+    Returns:
+        relative_diff: dictionary mapping keys like (0, 'Jxx') to waterfalls of relative
+            differences between old and new gains
+        avg_relative_diff: dictionary mapping antpols (e.g. 'Jxx') to waterfalls. Flagged
+            gains are excluded from the average; completely flagged times and channels are
+            replaced by 0s to match the convention of chi^2 above
+
+    """
+    if denom is None:
+        denom = old_gains
+    relative_diff = {ant: np.abs((old_gains[ant] - new_gains[ant]) / denom[ant]) for ant in new_gains}
```

Comment:
there are times when denom is likely to be zero. best to do a true divide and save ourselves the hassle of having to flag infs or nans later..

Reply:
In general, we shouldn't have 0s in the gains (flagged gains usually have 1s in the values). But in the case where we do, what does `np.true_divide` actually get us? My understanding is that it's exactly the same as `/` in python 3...

Reply:
if `a = [1, 2, 3]` and `b = [1, 0, 1]` then `np.true_divide(a, b, where=~np.isclose(b, 0))` returns `array([1., 0., 3.])` whereas `a/b` returns `array([ 1., inf,  3.])` and raises a dividebyzero warning.

Reply:
Sometimes the data can be zero, whereas the model is not zero but the data isn't flagged (typically at band edges), so the gains go to zero